### PR TITLE
feat: Add pools package

### PR DESCRIPTION
**Pull request by chefjackson**
### Summary
🎣🔗🕒

This pull request refactors the frontend project to use the `@pancakeswap/pools` package as a dependency, which handles the logic of fetching and transforming the pool data from different sources and chains. This improves the accuracy, consistency, and maintainability of the pool data and the time-related information displayed in the pool cards. It also removes some redundant code and files, and adds some configuration and testing changes to support the new dependency.

> _We are the pools of fire, we burn across the chains_
> _We use the `@pancakeswap/pools` package, we simplify the pains_
> _We fetch and transform the data, we use the timestamps_
> _We are the pools of fire, we are the champions_

### Walkthrough
*  Add `@pancakeswap/pools` package as a local dependency and alias, which provides the logic of fetching and transforming the pool data from different sources and chains ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-197cd8ca285a4abd2f21479e0bf6e36e90b08528fcd7f3bdbe8d1221897e377dR48), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8R43), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-e0daeed87867fc4fc6b0b1a9c2d52c9bfbd175cd1822e100f43300c4046484adR16))
*  Update the data structure of the pool objects, using timestamps instead of block numbers, and a single string to represent the contract address and type tag ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-866179ec9d915092cf06773dd27d5feb7717e75213c896817019551574643d8aL15-R15), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-866179ec9d915092cf06773dd27d5feb7717e75213c896817019551574643d8aL63-R64), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-866179ec9d915092cf06773dd27d5feb7717e75213c896817019551574643d8aL71-R73), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-866179ec9d915092cf06773dd27d5feb7717e75213c896817019551574643d8aL84-R83), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-866179ec9d915092cf06773dd27d5feb7717e75213c896817019551574643d8aL90-R91), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-866179ec9d915092cf06773dd27d5feb7717e75213c896817019551574643d8aL105-R104), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-866179ec9d915092cf06773dd27d5feb7717e75213c896817019551574643d8aL113-R112), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-1154ef933edf010b7b55ffa0ac375720d8f3274ab251630ea2230b7945e66dd7L18-R19), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-913dda55fc2b6572b1970c34f8f06c78e0c941686f3726bf779e22bb813840ffL56-R56), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-913dda55fc2b6572b1970c34f8f06c78e0c941686f3726bf779e22bb813840ffL105-R105), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-ff47d795a471d8cb14c04e358cbf99ea49476e4fc42dfdf3516fbbd3c9ade93eL68-R68), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-ff47d795a471d8cb14c04e358cbf99ea49476e4fc42dfdf3516fbbd3c9ade93eL156-R161), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-ff47d795a471d8cb14c04e358cbf99ea49476e4fc42dfdf3516fbbd3c9ade93eL175-R171), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-ff47d795a471d8cb14c04e358cbf99ea49476e4fc42dfdf3516fbbd3c9ade93eL183-R178), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-7c6c252ff4c393eda11175b34e6af21c7db767dc1e3af58d2aab0deef6ce17ccL53-R95), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-dd335cac79a792b6be2dc88728874aa7bffece1439038ea7055760dbb83e8b7dL2-R69))
*  Update the pool contract interactions, using the `getPoolContractBySousId` function from the `@pancakeswap/pools` package instead of the `getSouschefContract` function ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-615c30d6c6f61b444636d64d6a14f50985cc9c40098057c21eaa564896f85851L73-R73), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-e4c0879b3617130edb131741eba0816230cf6e87597ccfc6c98faf264ed9e50fL18), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-8fc36f90353c94a71f2d079d764035e4b9cf590e85992a96a1ca43bd6a88a56eL34-R37), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-0189a20b061b4c4cd5fd74abb3ff0a22a05b9f5adde9583eb11606cb589a5af4L62), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-0189a20b061b4c4cd5fd74abb3ff0a22a05b9f5adde9583eb11606cb589a5af4L163-R164))
*  Update the pool page rendering, using the `PoolsPage` component with a `chains` property to filter the supported chain ids ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-462c4318ca4d00d4b40049a292d3f574ccc38e83022e1baf41ab8a2764319e3aL1-R9), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-d9c385d3ac97d1f1afdd747c6e15b077ab59b74970c6624a193e8cb34c2fbff0L1-R9))
*  Update the menu item filtering, using the `SUPPORTED_CHAIN_IDS` constant from the `@pancakeswap/pools` package ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-e0daeed87867fc4fc6b0b1a9c2d52c9bfbd175cd1822e100f43300c4046484adL97-R98))
*  Update the vault APY calculation, using the constants from the `@pancakeswap/pools` package instead of the `config/constants/pools` file ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-86c4a24afd608d6fbfea469bf7f5250a3015e79aeceae316c060f8f924968597L11-R11))
*  Update the block timestamp caching, adding a conditional statement to fetch and store the initial block timestamp for each chain id ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-5ceb9d5e2a50941c88f358e2afea0e55f90ea64191a679bd6c697fe2e177fc47R23-R26), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-5ceb9d5e2a50941c88f358e2afea0e55f90ea64191a679bd6c697fe2e177fc47R86-R91))
*  Update the test cases, using the new functions and data structure from the `@pancakeswap/pools` package ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-7c6c252ff4c393eda11175b34e6af21c7db767dc1e3af58d2aab0deef6ce17ccL1-R4), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-7c6c252ff4c393eda11175b34e6af21c7db767dc1e3af58d2aab0deef6ce17ccL53-R95))
*  Remove unused code and files, such as the `poolsConfig` constant and the `getSouschefContract` function ([link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-dd335cac79a792b6be2dc88728874aa7bffece1439038ea7055760dbb83e8b7dL2-R69), [link](https://github.com/pancakeswap/pancake-frontend/pull/6600/files?diff=unified&w=0#diff-0189a20b061b4c4cd5fd74abb3ff0a22a05b9f5adde9583eb11606cb589a5af4L62))

### CODE REVIEW

File: packages/pools/package.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  "version": "0.0.1",
+  "sideEffects": false,
+  "private": true,
+  "scripts": {
+    "build": "yarn typechain && tsup",
+    "test": "vitest --run",
+    "typechain": "typechain --out-dir src/abis/types --target=ethers-v5 \"src/abis/*.json\"",
+    "postinstall": "yarn typechain"
+  },
+  "main": "./dist/index.js",
```
Comment (chef-jojo) [inline packages/pools/package.json:12]: Let's skip the build we can import tsx directly for internal pkg

File: apps/web/src/views/Migration/hook/V1/Pool/useFetchUserPools.ts [hunk 2]
```diff
@@ -17,25 +17,37 @@ export interface PoolsState {
...
+    },
+    userDataLoaded: false,
   },
-  userDataLoaded: false,
 }
 
 export const useFetchUserPools = (account) => {
   const { chainId } = useActiveChainId()
-  const [userPoolsData, setPoolsUserData] = useState<PoolsState>(initialData)
+  const [userPoolsData, setPoolsUserData] = useState<PoolsState>(initialData[chainId || 56])
```
Comment (ChefMomota) [inline apps/web/src/views/Migration/hook/V1/Pool/useFetchUserPools.ts:50]: `56` -> `ChainId.BSC`

Comment (chefjackson) [reply apps/web/src/views/Migration/hook/V1/Pool/useFetchUserPools.ts]: https://github.com/pancakeswap/pancake-frontend/pull/6600/commits/fcb1ae2ff2599a1cc6de50f61c9169de01ee9c8a

File: packages/pools/tsup.config.ts
Comment (chef-jojo) [inline packages/pools/tsup.config.ts:1]: can remove this file

File: packages/uikit/src/widgets/Pool/types.ts
Comment (chef-jojo) [inline packages/uikit/src/widgets/Pool/types.ts:1]: Let's plan to move the widget Pool into Pool in the future. Right now every farms/pool package will trigger uikit build and retrigger all apps build

File: packages/pools/tsup.config.ts
Comment (chefjackson) [reply packages/pools/tsup.config.ts:1]: Removed
https://github.com/pancakeswap/pancake-frontend/pull/6600/commits/a7a4ac0ae43ee39ed1499ea888fdbce037c3c20f

File: packages/uikit/src/widgets/Pool/types.ts
Comment (chefjackson) [reply packages/uikit/src/widgets/Pool/types.ts:1]: I'm thinking splitting the uikit package into several small ui packages

Comment (chefjackson) [reply packages/uikit/src/widgets/Pool/types.ts:1]: but yeah need to at least move the widget out of the uikit package

File: apps/web/src/state/pools/helpers.ts [hunk 3]
```diff
@@ -42,29 +42,29 @@ const transformProfileRequirement = (profileRequirement?: { required: boolean; t
...
     profileRequirement,
-    startBlock,
+    startTimestamp,
     ...rest
   } = pool
 
   return {
     ...rest,
-    startBlock,
+    startTimestamp,
```
Comment (chefilip) [inline apps/web/src/state/pools/helpers.ts:60]: hmm. For BSC, it should be block-based caluclation instead of timestamp, right?

Comment (chefjackson) [reply apps/web/src/state/pools/helpers.ts:60]: Yes and it's already converted from block to timestamp on bsc

Comment (chefjackson) [reply apps/web/src/state/pools/helpers.ts:60]: Can refer https://github.com/pancakeswap/pancake-frontend/pull/6600/files#diff-5d7737987d8acc78406f7c7c03dafcaf248abeba85b13dfd87d08a169834cc09R124

**Review by chef-jojo — APPROVED** [review]

File: apps/web/src/state/info/utils.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { ChainId } from '@pancakeswap/sdk'
+
+import { CHAIN_QUERY_NAME } from 'config/chains'
+
+import { multiChainPaths } from './constant'
+
+export function getTokenInfoPath(chainId: ChainId, address: string) {
+  return `/info${multiChainPaths[chainId]}/tokens/${address}?chain=${CHAIN_QUERY_NAME[chainId]}`
```
Comment (Chef-Yogi) [inline apps/web/src/state/info/utils.ts:8]: info/v3/tokens/0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82

Comment (Chef-Yogi) [reply apps/web/src/state/info/utils.ts:8]: shall we switch it to the v3 info thanks a lot

**Review by ChefMomota — APPROVED** [review]

**Pull request merged**